toolkit/tools/pkg/imagecustomizerlib/imagecustomizer.go

```diff
@@ -285,13 +287,16 @@ func CustomizeImage(buildDir string, baseConfigPath string, config *imagecustomi
 	}
 
 	if config.Output.Artifacts != nil {
-		outputDir := file.GetAbsPathWithBase(baseConfigPath, config.Output.Artifacts.Path)
-
-		err = outputArtifacts(config.Output.Artifacts.Items, outputDir,
-			imageCustomizerParameters.buildDirAbs, imageCustomizerParameters.rawImageFile)
+		outputArtifactsMetadata, err := outputArtifacts(config.Output.Artifacts.Items, config.Output.Artifacts.Path,
+			imageCustomizerParameters.buildDirAbs, imageCustomizerParameters.rawImageFile, baseConfigPath)
 		if err != nil {
 			return err
 		}
+
+		err = writeInjectFilesYaml(outputArtifactsMetadata, baseConfigPath)
```

Comment:
I think it would be cleaner if `writeInjectFilesYaml` was called in `outputArtifacts`.